### ISSUE TITLE
Analyses page fails to open (freezes indefinitly)

### ISSUE THREAD

**Issue by apetkau** [phac-nml/irida#1004]
## Describe the bug

The **Analyses > Your Analyses** page freezes up and fails to load any analyses pipelines. I get an error:

```
19 May 2021 02:57:01,260 ERROR ca.corefacility.bioinformatics.irida.ria.web.errors.ExceptionHandlerController:117 - No enum constant ca.corefacility.bioinformatics.
irida.model.enums.AnalysisState.DOWNLOADING; nested exception is java.lang.IllegalArgumentException: No enum constant ca.corefacility.bioinformatics.irida.model.enu
ms.AnalysisState.DOWNLOADING
org.springframework.dao.InvalidDataAccessApiUsageException: No enum constant ca.corefacility.bioinformatics.irida.model.enums.AnalysisState.DOWNLOADING; nested exce
ption is java.lang.IllegalArgumentException: No enum constant ca.corefacility.bioinformatics.irida.model.enums.AnalysisState.DOWNLOADING
        at org.springframework.orm.jpa.EntityManagerFactoryUtils.convertJpaAccessExceptionIfPossible(EntityManagerFactoryUtils.java:374)
        at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:257)
        at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:528)
        at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:61)
        at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:242)
        at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:153)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.invoke(CrudMethodMetadataPo
stProcessor.java:178)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:93)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:212)
        at com.sun.proxy.$Proxy210.findRevisions(Unknown Source)
        at ca.corefacility.bioinformatics.irida.ria.web.analysis.auditing.AnalysisAudit.getAnalysisRunningTime(AnalysisAudit.java:41)
        at ca.corefacility.bioinformatics.irida.ria.web.analysis.AnalysesTableAjaxController.createAnalysisModel(AnalysesTableAjaxController.java:228)
...
``` 

## Steps to reproduce the problem

It's a bit more difficult to reproduce this problem because I believe the issue exists in only a specific IRIDA instance. But in general:
1. Login to IRIDA
2. Go to **Analyses > My Analyses**
3. The page does not load

## Expected behaviour
I expected the page to load.

### Frozen My Analyses page
![image](https://user-images.githubusercontent.com/200517/118777064-489ada80-b84e-11eb-9bb8-9ba06a591fb5.png)

**Comment by apetkau** [comment]
Looking through the code in [AnalysisState.java](https://github.com/phac-nml/irida/blob/21.01.4/src/main/java/ca/corefacility/bioinformatics/irida/model/enums/AnalysisState.java) it is true that `DOWNLOADING` does not exist. This was removed back in 2017: https://github.com/phac-nml/irida/commit/753c7aaf2a6934def276bd2fd290872ba9446068#diff-50a6b1679b6567c469d6674e104dc56847f749bf6f2120067b1280ae7921f514

However, looking through the specific line of code in the stack trace that triggered this exception:

https://github.com/phac-nml/irida/blob/572c660d802755d1c93c4e4ff3e482495d075e9e/src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/auditing/AnalysisAudit.java#L41-L42

It is trying to load up old `AnalysisSubmission` objects from the audit tables. I suspect what's going on is that this IRIDA instance has old `AnalysisSubmission` records in the audit tables that have the `DOWNLOADING` state, and since it no longer exists it triggers this exception.

**Comment by apetkau** [comment]
I suspect this issue could be reproduced by switching to an IRIDA version that still has the `DOWNLOADING` state, getting an analysis stuck in the `DOWNLOADING` state, and then upgrading IRIDA.

**Comment by apetkau** [comment]
Fixed in #1006
